f5/bigip/resource.py

```diff
@@ -295,6 +305,37 @@ def _handle_requests_params(self, kwargs):
             requests_params.update({'params': params})
         return requests_params
 
+    def _check_exclusive_parameters(self, **kwargs):
+        """Check for mutually exclusive attributes in kwargs.
+
+        :raises ExclusiveAttributesPresent
+        """
+        if len(self._meta_data['exclusive_attributes']) > 0:
+            attr_set = set(kwargs.keys())
+            ex_set = set(self._meta_data['exclusive_attributes'][0])
+            common_set = attr_set.intersection(ex_set)
+            if len(common_set) > 1:
+                cset = ', '.join(common_set)
+                error = 'Mutually exclusive arguments submitted. ' \
+                        'The following arguments cannot be set ' \
+                        'together: "%s".' % cset
+                raise ExclusiveAttributesPresent(error)
+
+    @staticmethod
+    def _check_required_parameters(rqset, **kwargs):
+        """Helper function to do operation on sets
+
+        ::returns bool and variable
+        """
+        key_set = set(kwargs.keys())
+        required_minus_received = rqset - key_set
+        if required_minus_received != set():
```

Comment:
The idea is that this method is merely checking that things are all good, and it finds that or it doesn't. Getting a return value of True, may not mean much to a user, but raising an exception will mean that something's not right. I think it's just a little too specific of a use here and could be generalized a bit to raise. The other thing I'm thinking about is that it really has no information to return back to the caller that can't be done with an exception.